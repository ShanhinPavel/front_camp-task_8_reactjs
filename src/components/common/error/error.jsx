import React from 'react';

export const Error = () => (
  <div style={{ margin: '50vh auto', color: 'white', textAlign: 'center' }}>
    <h1>Oops! Something went wrong.</h1>
    <p>Please reload the page.</p>
  </div>
);
