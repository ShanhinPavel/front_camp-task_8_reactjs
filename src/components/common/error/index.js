export { ErrorBoundary } from './error-boundary';
export { Error } from './error';
