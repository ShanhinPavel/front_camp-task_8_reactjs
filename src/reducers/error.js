export const error = (state = false, action) => {
  switch (action.type) {
    case 'DOWNLOADING_ERROR':
      return action.error;
    default:
      return state;
  }
};
