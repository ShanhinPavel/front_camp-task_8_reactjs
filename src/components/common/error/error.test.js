import React from 'react';
import { shallow, mount } from 'enzyme';
import { create } from 'react-test-renderer';

import { Error } from './error';

describe('Render component', () => {
  it('Component should render', () => {
    const renderedValue = create(<Error />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
