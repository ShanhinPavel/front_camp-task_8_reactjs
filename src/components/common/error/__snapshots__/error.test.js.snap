// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component Component should render 1`] = `
<div
  style={
    Object {
      "color": "white",
      "margin": "50vh auto",
      "textAlign": "center",
    }
  }
>
  <h1>
    Oops! Something went wrong.
  </h1>
  <p>
    Please reload the page.
  </p>
</div>
`;
